File: android/app/src/main/java/io/ionic/starter/IPluginEventHandler.java
package io.ionic.starter;

import com.vidyo.VidyoClient.Endpoint.Participant;

public interface IPluginEventHandler {

    void onInitialized(boolean status);

    void onConnected();

    void onDisconnected(String reason);

    void onFailure(String reason);

    void onParticipantJoined(Participant participant);

    void onParticipantLeft(Participant participant);
}

File: ios/App/App/IPluginEventHandler.swift
//
//  IPluginEventHandler.swift
//  App
//
//  Created by taras.melko on 08.06.2021.
//

import Foundation

protocol IPluginEventHandler: class {

    func onInitialized(status: Bool)

    func onConnected()

    func onDisconnected(reason: String)

    func onFailure(reason: String)

    func onParticipantJoined(participant: VCParticipant)

    func onParticipantLeft(participant: VCParticipant)
}
